File: 1309-decrypt-string-from-alphabet-to-integer-mapping/1309-decrypt-string-from-alphabet-to-integer-mapping.java
class Solution {
    public String freqAlphabets(String s) {
        int l = s.length();
	    StringBuilder sb = new StringBuilder();
	    while(l>0) {
	        int temp;
	        if(s.charAt(l-1) == '#') {
	            temp = Integer.parseInt(s.substring(l-3,l-1))+96;
	            sb.append((char)temp);
	            l-=3;
	        } else {
	            temp = Character.getNumericValue(s.charAt(l-1))+96;
	            sb.append((char)temp);
	            l--;        
	        }
	    }
	    return sb.reverse().toString();
    }
}

File: 1309-decrypt-string-from-alphabet-to-integer-mapping/1309-decrypt-string-from-alphabet-to-integer-mapping.py
class Solution:
    def freqAlphabets(self, s: str) -> str:
        l = len(s)
        output = ""
        while(l>0):
            if s[l-1] == '#':
                output+=chr(int(s[l-1-2: l-1])+96)
                l-=3
            else:
                output+=chr(int(s[l-1])+96)
                l-=1
        return output[::-1]